sematic/ui/packages/common/src/pages/RunDetails/index.tsx

```diff
@@ -29,16 +38,49 @@ const RunDetails = (props: RunDetailsProps) => {
         return <NotesPane />;
     }, []);
 
-    const contextValue = useMemo(() => {
-        return {
-            rootRun,
-            resolution
-        };
-    }, [rootRun, resolution]);
+    const selectedRun = useMemo(() => {
+        let runId = selectedRunIdHash;
+        if (!runId) {
+            runId = rootRun.id;
+        }
+        if (!graph) {
+            return undefined;
+        }
+        return graph.runsById.get(runId) || rootRun;
+    }, [selectedRunIdHash, graph, rootRun]);
+
+    const rootRunContextValue = useMemo(() => ({
+        rootRun,
+        resolution,
+        graph,
+        isResolutionLoading,
+        isGraphLoading
+    }), [rootRun, resolution, graph, isResolutionLoading, isGraphLoading]);
+
+    const runDetailsSelectionContext = useMemo(() => ({
+        selectedRun,
+        setSelectedRunId
+    }), [selectedRun, setSelectedRunId]);
+
+    const updateHash = useHashUpdater();
+
+    // Clear the `run` hash when the selected run is not available in the graph or
+    // the selected run is the root run.
+    useEffect(() => {
+        if (selectedRunIdHash === rootRun.id ||
+            (!!graph && !graph.runsById.get(selectedRunIdHash))) {
+            if (!!selectedRunIdHash) {
+                updateHash({ "run": RESET }, true);
+            }
+            return;
+        }
+    }, [selectedRunIdHash, graph, rootRun, updateHash]);
 
-    return <RootRunContext.Provider value={contextValue}>
-        <ThreeColumns onRenderLeft={onRenderLeft} onRenderCenter={onRenderCenter}
-            onRenderRight={onRenderRight} />
+    return <RootRunContext.Provider value={rootRunContextValue}>
+        <RunDetailsSelectionContext.Provider value={runDetailsSelectionContext}>
+            <ThreeColumns onRenderLeft={onRenderLeft} onRenderCenter={onRenderCenter}
+                onRenderRight={onRenderRight} />
```

Comment:
When resizing the window to make it smaller, the Execution Graph panel scroll bar appears on the left, and minimized to the point where it is unusable, but rendered. I had trouble understanding what that grey dot was when I first saw it. Please consider disabling it if there is no room to scroll.

<img width="310" alt="image" src="https://github.com/sematic-ai/sematic/assets/1894533/48f51f75-3ea1-43a3-9d5e-b24aae163dd7">

<img width="312" alt="image" src="https://github.com/sematic-ai/sematic/assets/1894533/63ca0b13-6168-4456-b746-ad0f0c6b03b3">

Reply:
Thanks for mentioning it. It is a known issue. @neutralino1 and I planned to collapse the other sections when there was no room to show the tree. It will be a dedicated PR later to resolve this issue comprehensively.